src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -334,11 +335,14 @@ double SolverInterfaceImpl::initialize()
     context.initializeWaveform();
   }
 
+  // Needs to be removed for test Integration/Serial/Whitebox/TestExplicitWithDataScaling
+  /**
   if (_couplingScheme->hasDataBeenReceived()) {
     performDataActions({action::Action::READ_MAPPING_PRIOR}, 0.0, 0.0, 0.0, dt);
     mapReadData();
     performDataActions({action::Action::READ_MAPPING_POST}, 0.0, 0.0, 0.0, dt);
   }
+  */
```

Comment:
Commenting out this part of the code actually makes this PR breaking.

Reply:
I played around with this part a bit yesterday: If I comment it out, calling `initializeData()` in every test is unproblematic, but if I revert some tests to the state from `develop`, then they break. There might be a way to make this PR non-breaking, but I think this is complicated and not really worth it. Therefore, I will directly build up on this PR and merge `initialize()` and `initializeData()` now. This is also related to #1349 and https://github.com/precice/precice/pull/1029 (waveform and serial coupling).